lib/workers/repository/model/commit-message.ts

```diff
@@ -1,54 +1,83 @@
-export class CommitMessage {
-  public static readonly SEPARATOR: string = ':';
+import type { CommitMessageJSON } from '../../../types';
 
-  private message = '';
+/**
+ * @see https://git-scm.com/docs/git-commit#_discussion
+ *
+ * [optional prefix]: <suject>
+ * [optional body]
+ * [optional footer]
+ */
+export abstract class CommitMessage {
+  static readonly #SEPARATOR: string = ':';
+  static readonly #EXTRA_WHITESPACES = /\s+/g;
 
-  private prefix = '';
-
-  constructor(message = '') {
-    this.setMessage(message);
-  }
+  #body = '';
+  #footer = '';
+  #subject = '';
 
   public static formatPrefix(prefix: string): string {
     if (!prefix) {
       return '';
     }
 
-    if (prefix.endsWith(CommitMessage.SEPARATOR)) {
+    if (prefix.endsWith(CommitMessage.#SEPARATOR)) {
       return prefix;
     }
 
-    return `${prefix}${CommitMessage.SEPARATOR}`;
+    return `${prefix}${CommitMessage.#SEPARATOR}`;
   }
 
-  public setMessage(message: string): void {
-    this.message = (message || '').trim();
+  toJSON(): CommitMessageJSON {
+    return {
+      body: this.#body,
+      footer: this.#footer,
+      subject: this.#subject,
+    };
   }
 
-  public setCustomPrefix(prefix?: string): void {
-    this.prefix = (prefix ?? '').trim();
+  toString(): string {
+    const parts: ReadonlyArray<string | undefined> = [
+      this.title,
+      this.#body,
+      this.#footer,
+    ];
+
+    return parts.filter(Boolean).join('\n\n');
   }
 
-  public setSemanticPrefix(type?: string, scope?: string): void {
-    this.prefix = (type ?? '').trim();
+  get title(): string {
+    return [CommitMessage.formatPrefix(this.prefix), this.formatSubject()]
+      .join(' ')
+      .trim();
+  }
 
-    if (scope?.trim()) {
-      this.prefix += `(${scope.trim()})`;
-    }
+  set body(value: string) {
+    this.#body = value.trim();
   }
 
-  public toString(): string {
-    const prefix = CommitMessage.formatPrefix(this.prefix);
-    const message = this.formatMessage();
+  set footer(value: string) {
+    this.#footer = value.trim();
+  }
 
-    return [prefix, message].join(' ').trim();
+  set subject(value: string) {
+    this.#subject = value.trim();
```

Comment:
```suggestion
    this.#subject = value?.trim();
```
otherwise that operator in next line makes no sense. 😉